src/gui/sdlmain.cpp

```diff
@@ -3305,19 +3194,61 @@ static SDL_Rect calc_viewport_fit(const int win_w, const int win_h)
 	assert(std::isfinite(sdl.draw.scaley));
 
 	// limit the window to the user's desired viewport, if configured
-	const auto [lwin_w, lwin_h] = restrict_to_viewport_resolution(win_w, win_h);
-
-	// calculate the aspect ratios of the draw buffer and the limited window
-	const auto draw_aspect = (sdl.draw.width * sdl.draw.scalex) /
-	                         (sdl.draw.height * sdl.draw.scaley);
-	const auto lwin_aspect = static_cast<double>(lwin_w) / lwin_h;
-
-	// calculate the viewport contingent on the aspect ratio of the limited
-	// window versus draw buffer
-	const auto lwin_is_wider = lwin_aspect > draw_aspect;
-	const auto view_w = lwin_is_wider ? iround(lwin_h * draw_aspect) : lwin_w;
-	const auto view_h = lwin_is_wider ? lwin_h : iround(lwin_w / draw_aspect);
+	const auto [bounds_w,
+	            bounds_h] = restrict_to_viewport_resolution(win_w, win_h);
+
+	// Calculate the aspect ratio of the emulated display mode.
+	const auto output_aspect = (sdl.draw.width * sdl.draw.scalex) /
+	                           (sdl.draw.height * sdl.draw.scaley);
+
+	int view_w = 0;
+	int view_h = 0;
+	switch (sdl.integer_scaling_mode) {
+	case IntegerScalingMode::Off: {
+		// Calculate the viewport contingent on the aspect ratio of the
+		// viewport bounds versus display mode.
+		const auto bounds_aspect = static_cast<double>(bounds_w) / bounds_h;
+		if (bounds_aspect > output_aspect) {
+			view_w = iround(bounds_h * output_aspect);
+			view_h = bounds_h;
+		} else {
+			view_w = bounds_w;
+			view_h = iround(bounds_w / output_aspect);
+		}
+		break;
+	}
+	case IntegerScalingMode::Horizontal: {
+		// Calculate scaling multiplier that will allow to fit the whole
+		// image into the window or viewport bounds.
+		auto integer_scale_factor = std::min(
+		        bounds_w / sdl.draw.width,
+		        static_cast<int>(bounds_h / (sdl.draw.height *
+		                                     sdl.draw.pixel_aspect)));
+		// Safeguard to not let the image implode into a black hole.
+		if (integer_scale_factor < 1)
+			integer_scale_factor = 1;
```

Comment:
Please enclose single statements in curly braces too (reason: it's very error prone when you add more lines to it later).

Reply:
```
# Commenting out for now as VisualStudio 2019 doesn't recognise it
#InsertBraces: true
```

...oh. That's why.

Reply:
Minor thing, but bump 😄

Reply:
GitHub doesn't update these previews, better check commits directly.